### PR TITLE
Feature 152 remove index requirement from datasets

### PR DESCRIPTION
**Pull request by j48zheng**

### CODE REVIEW

File: holoclean/dataengine.py [hunk 1]
```diff
@@ -411,6 +413,9 @@ def ingest_data(self, filepath, dataset):
         # Spawn new reader and load data into dataframe
         file_reader = Reader(self.holo_env.spark_session)
         df = file_reader.read(filepath)
+        if GlobalVariables.index_name in df.schema.names:
+            raise Exception("attribute + " + GlobalVariables.index_name + " not allowed in dataset")
+        df = df.select("*").withColumn(GlobalVariables.index_name, monotonically_increasing_id() + 1)
```
Comment (thodrek) [inline holoclean/dataengine.py:418]: will this start the Id's from 1 or 0? Make sure we are consistent with what we want. Don't remember if we wanted everything to start with 0 or 1.

File: holoclean/errordetection/mysql_dcerrordetector.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 from holoclean.utils.dcparser import DCParser
 from errordetector import ErrorDetection
+from ..global_variables import GlobalVariables
```
Comment (thodrek) [inline holoclean/errordetection/mysql_dcerrordetector.py]: why do we need .. and not holoclean. ?

File: holoclean/featurization/dcfeaturizer.py [hunk 3]
```diff
@@ -1,5 +1,6 @@
 from featurizer import Featurizer
 from pyspark.sql.types import StructField, StructType, StringType, IntegerType
+from ..global_variables import GlobalVariables
```
Comment (thodrek) [inline holoclean/featurization/dcfeaturizer.py]: same as previous comment.

File: holoclean/learning/accuracy.py [hunk 4]
```diff
@@ -1,3 +1,6 @@
+from ..global_variables import GlobalVariables
```
Comment (thodrek) [inline holoclean/learning/accuracy.py]: same as previous comment. holoclean.global_variables

File: holoclean/learning/accuracy.py [hunk 5]
```diff
@@ -50,9 +53,9 @@ def _flatting(self):
         counter = 0
 
         for rv_attr in rv_attrs:
-            if rv_attr != "index" and rv_attr != "Index":
+            if rv_attr != GlobalVariables.index_name:
                 query_for_flattening = """ (SELECT \
-                                       DISTINCT t1.index as tid,'""" + \
+                                       DISTINCT t1.__ind as tid,'""" + \
```
Comment (thodrek) [inline holoclean/learning/accuracy.py:58]: did you make sure that this query runs fine?

File: holoclean/utils/dcparser.py [hunk 6]
```diff
@@ -1,3 +1,6 @@
+from ..global_variables import GlobalVariables
```
Comment (thodrek) [inline holoclean/utils/dcparser.py]: Same as before.

File: holoclean/utils/pruning.py [hunk 7]
```diff
@@ -1,4 +1,5 @@
 from pyspark.sql.types import *
+from ..global_variables import GlobalVariables
```
Comment (thodrek) [inline holoclean/utils/pruning.py]: same as before.

File: test/Unit_tests/unittest_dcfeaturizer.py [hunk 8]
```diff
@@ -4,6 +4,7 @@
 from holoclean.holoclean import HoloClean, Session
 from holoclean.errordetection.mysql_dcerrordetector import MysqlDCErrorDetection
 from holoclean.featurization.dcfeaturizer import SignalDC
+from holoclean.global_variables import GlobalVariables
```
Comment (thodrek) [inline test/Unit_tests/unittest_dcfeaturizer.py:7]: This is correct :)

**Review by thodrek — CHANGES_REQUESTED** [review]
Please fix the way global_variables are imported.

File: holoclean/dataengine.py (hunk 1 above)
Comment (j48zheng) [reply holoclean/dataengine.py:418]: We want to start from 1, which we made sure it did.

File: holoclean/learning/accuracy.py (hunk 5 above)
Comment (j48zheng) [reply holoclean/learning/accuracy.py:58]: yes

File: holoclean/dataengine.py (hunk 1 above)
Comment (thodrek) [reply holoclean/dataengine.py:418]: Sounds good.

**Review by thodrek — APPROVED** [review]

**Pull request merged**